commcare-cloud/commcare_cloud.py

```diff
@@ -44,30 +51,50 @@ def make_parser(parser):
     @staticmethod
     def run(args, unknown_args):
         check_branch(args)
+        public_vars = get_public_vars(args.environment)
+        pem = public_vars.get('commcare_cloud_pem', None)
+        strict_host_key_checking = public_vars.get('commcare_cloud_strict_host_key_checking', True)
+        ask_vault_pass = public_vars.get('commcare_cloud_use_value', True)
 
-        def ansible_playbook(environment, playbook, vault_password, *cmd_args):
+        def ansible_playbook(environment, playbook, *cmd_args):
             cmd_parts = (
                 'ANSIBLE_CONFIG={}'.format(os.path.expanduser('~/.commcare-cloud/ansible/ansible.cfg')),
                 'ansible-playbook',
                 os.path.expanduser('~/.commcare-cloud/ansible/{playbook}'.format(playbook=playbook)),
-                '-u', 'ansible',
                 '-i', os.path.expanduser('~/.commcare-cloud/inventory/{env}'.format(env=environment)),
                 '-e', '@{}'.format(os.path.expanduser('~/.commcare-cloud/vars/{env}/{env}_vault.yml'.format(env=environment))),
                 '-e', '@{}'.format(os.path.expanduser('~/.commcare-cloud/vars/{env}/{env}_public.yml'.format(env=environment))),
-                '--vault-password-file=/bin/cat',
                 '--diff',
             ) + cmd_args
+
+            if '-u' not in unknown_args:
+                cmd_args += ('-u', 'ansible')
+
+            if ask_vault_pass:
+                cmd_parts += ('--vault-password-file=/bin/cat',)
+
+            common_ssh_args = []
+            if pem:
+                common_ssh_args.extend(['-i', pem])
+            if not strict_host_key_checking:
+                common_ssh_args.append('-o StrictHostKeyChecking=no')
```

Comment:
This is what we get when a provider cycles IPs right? Do we want this around long term? We should probably start up a repository for the hashes we can trust

Reply:
I added this as a workaround for easy bootstraping on AWS for temporary environments. I agree that even better would be proactively fetch and whitelist the fingerprints, but I wasn't finding an easy way to do that on my first round of quick googling, so I went with this. Agree that it'd be nice to have a better long-term way to manage this for our prod VMs.

Reply:
Yeah. that makes sense. I was thinking about htis when having to deal with the new icds cluster. Maybe there's a way we could deploy ~/.known_hosts with a static file? I don't think host keys are private (you get prompted with the host key before authentication), so there shouldn't be any privacy issues with that